container/factory.go

```diff
@@ -89,30 +86,27 @@ func HasFactories() bool {
 }
 
 // Create a new ContainerHandler for the specified container.
-func NewContainerHandler(name string, inHostNamespace bool) (ContainerHandler, bool, error) {
+func NewContainerHandler(name string, inHostNamespace bool) (ContainerHandler, error) {
 	factoriesLock.RLock()
 	defer factoriesLock.RUnlock()
 
 	// Create the ContainerHandler with the first factory that supports it.
 	for _, factory := range factories {
-		canHandle, canAccept, err := factory.CanHandleAndAccept(name)
+		handler, ignore, err := factory.NewContainerHandler(name, inHostNamespace)
 		if err != nil {
 			glog.V(4).Infof("Error trying to work out if we can handle %s: %v", name, err)
 		}
-		if canHandle {
-			if !canAccept {
-				glog.V(3).Infof("Factory %q can handle container %q, but ignoring.", factory, name)
-				return nil, false, nil
-			}
-			glog.V(3).Infof("Using factory %q for container %q", factory, name)
-			handle, err := factory.NewContainerHandler(name, inHostNamespace)
-			return handle, canAccept, err
+		if handler != nil {
+			return handler, err
+		} else if ignore {
```

Comment:
no, if we want to ignore this container then should return a nil handler.  ignore is only if we return a nil handler to tell the factory if it should ignore this container or continue checking the other factories

i.e. ignore is the equivalent of handle = true, accept = false in the previous code.